source

```diff
@@ -107224,6 +107259,121 @@ document.querySelector("button").addEventListener("click", bound);
+  options</span> <var>options</var>:</p>
+
+  <ol>
+   <li><p>If <span data-x="concept-environment-noscript">scripting is disabled</span> for
+   <var>settings</var>, then set <var>bodyBytes</var> to the byte sequence 0x00 0x61 0x73 0x6d 0x01
+   0x00 0x00 0x00.</p></li>
+
+   <li><p>Let <var>script</var> be a new <span>module script</span> that this algorithm will
+   subsequently initialize.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-settings-object">settings
+   object</span> to <var>settings</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-base-url">base URL</span> to
+   <var>baseURL</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-script-fetch-options">fetch
+   options</span> to <var>options</var>.</p></li>
+
+   <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span> and
+   <span data-x="concept-script-error-to-rethrow">error to rethrow</span> to null.</p></li>
+
+   <li>
+    <p>Let <var>result</var> be <span data-x="js-ParseModule">parse a WebAssembly module</span>(<var>source</var>,
+    <var>settings</var>'s <span data-x="environment settings object's realm">realm</span>,
+    <var>script</var>).</p>
+
+    <p class="note">Passing <var>script</var> as the last parameter here ensures
+    <var>result</var>.[[HostDefined]] will be <var>script</var>.</p>
+   </li>
+
+   <li>
+    <p>If <var>result</var> is a <span>list</span> of errors, then:</p>
+
+    <ol>
+     <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span> to
+     <var>result</var>[0].</p></li>
+
+     <li><p>Return <var>script</var>.</p></li>
+    </ol>
+   </li>
+
+   <li id="validate-requested-module-specifiers">
+    <p><span data-x="list iterate">For each</span> <span>ModuleRequest record</span>
+    <var>requested</var> of <var>result</var>.[[RequestedModules]]:</p>
+
+    <ol>
+     <li>
+     <p>If <var>requested</var>.[[Attributes]] contains a <span>Record</span> <var>entry</var>
+     such that <var>entry</var>.[[Key]] is not "<code data-x="">type</code>", then:</p>
+
+     <ol>
+      <li><p>Let <var>error</var> be a new <code>SyntaxError</code> exception.</p></li>
+
+      <li><p>Set <var>script</var>'s <span data-x="concept-script-parse-error">parse error</span>
+      to <var>error</var>.</p></li>
+
+      <li><p>Return <var>script</var>.</p></li>
+     </ol>
+
+     <p class="note">The JavaScript specification re-performs this validation when loading
+     <var>script</var>'s dependencies. It is duplicated here to skip unnecessary work given that,
+     in case of an invalid attribute key, loading the module graph would fail anyway. Additionally,
+     this makes the errors reported for invalid static imports consistent with dynamic imports, for
+     which invalid attributes are reported before calling into HTML and thus before validating the
+     imported specifier.</p>
+     </li>
```

Comment:
Maybe we can replace all of this with a note saying that WebAssembly modules cannot currently define attributes for their imports.

Reply:
Done.